src/Calculator.jsx

```diff
@@ -143,47 +159,109 @@ const Numbers = styled.section`
   }
 `;
 
-const Calculator = () => (
-  <Container>
-    <SubTitle>
-      Please select your daily activity and choose the amout by clicking on it again
-    </SubTitle>
-    <Activitys>
-      <IconContainer>
-        <Icon>
-          <Img src="/bottle.png" alt="bottle" />
-        </Icon>
-        <Icon>
-          <Img src="/glass.png" alt="glass" />
-        </Icon>
-        <Icon>
-          <Img src="/flush.png" alt="flush" />
-        </Icon>
-      </IconContainer>
-      <IconContainer>
-        <Icon>
-          <Img src="/washing-hands.png" alt="washing hands" />
-        </Icon>
-        <Icon>
-          <Img src="/shower.png" alt="shower" />
-        </Icon>
-        <Icon>
-          <Img src="/plumbing-pipe.png" alt="plumbing pipe" />
-        </Icon>
-      </IconContainer>
-      <Calculate>CALCULATE</Calculate>
-    </Activitys>
-    <ResultsContainer>
-      <Results>
-        <ResultsTitle>Your cost</ResultsTitle>
-        <Numbers>5</Numbers>
-      </Results>
-      <Results>
-        <ResultsTitle>Liters</ResultsTitle>
-        <Numbers>5</Numbers>
-      </Results>
-    </ResultsContainer>
-  </Container>
-);
+// Functionality
 
+class Calculator extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      userChoices: {
+        Toilet: 0,
+        Shower: 0,
+        'Faucet tap': 0,
+        'Washing hands': 0,
+        Cup: 0,
+        Bottle: 0,
+      },
+      cost: 0,
+      liters: 0,
+      shows: {
+        Toilet: 0,
+        Shower: 0,
+        'Faucet tap': 0,
+        'Washing hands': 0,
+        Cup: 0,
+        Bottle: 0,
+      },
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleClick(userChoice) {
+    this.setState({
+      userChoices: {
+        ...this.state.userChoices,
+        [userChoice]: this.state.userChoices[userChoice] + 1,
+      },
+      shows: {
+        ...this.state.shows,
+        [userChoice]: '1',
+      },
+    });
+  }
+
+  handleSubmit() {
+    const { cost, liters } = calculatorLogic(data, this.state.userChoices);
+    this.setState({ cost, liters });
+  }
+
+  // App
+
+  render() {
+    return (
+      <Container>
+        <SubTitle>
+          Please select your daily activity and choose the amout by clicking on it again
+        </SubTitle>
+        <Activitys>
+          <IconContainer>
+            <Icon onClick={() => this.handleClick('Bottle')}>
```

Comment:
Would it be worth having this as a map to return the Icons to the render? lots of repeated code here in a v big file?
Or just pulling out the Icon as a separate component?

I would definitely think about some refactoring now the file seems v big and hard to read. You could just move out the styles but also think about changing some of these to presentational components and passing things down as props?

Reply:
hmmmm yeah having the icons in an array and mapping through them would be ideal but for now i guess since it's an MVP its fine, and yes about they styles this is a priority i would say cause almost every component has a `Container` or `Ttitle` atleast. but i'll guess will do it later on